mpu6050/mpu6050.go

```diff
@@ -1,95 +1,232 @@
+// It has four possible values +- 250°/s, 500°/s, 1000°/s, 2000°/s.
+// The function takes values of gyroRange from 0-3 where 0 means the
+// lowest FSR (250°/s) and 3 is the highest FSR (2000°/s).
+func (p *Device) SetRangeGyro(gyroRange byte) (err error) {
+	switch gyroRange {
+	case GYRO_RANGE_250:
+		p.gRange = 250
+	case GYRO_RANGE_500:
+		p.gRange = 500
+	case GYRO_RANGE_1000:
+		p.gRange = 1000
+	case GYRO_RANGE_2000:
+		p.gRange = 2000
+	default:
+		return errors.New("invalid gyroscope FSR input")
+	}
+	// setFullScaleGyroRange
+	var gConfig [1]byte
+
+	if err = p.read(_GYRO_CONFIG, gConfig[:]); err != nil {
+		return err
+	}
+	gConfig[0] = (gConfig[0] & (^_G_FS_SEL)) | (gyroRange << _G_FS_SHIFT) // Escribo solo el campo de FS_sel
+
+	if err = p.write8(_GYRO_CONFIG, gConfig[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetRangeAccel configures the full scale range of the accelerometer.
+// It has four possible values +- 2g, 4g, 8g, 16g.
+// The function takes values of accRange from 0-3 where 0 means the
+// lowest FSR (2g) and 3 is the highest FSR (16g)
+func (p *Device) SetRangeAccel(accRange byte) (err error) {
+	switch accRange {
+	case ACCEL_RANGE_2:
+		p.aRange = 2
+	case ACCEL_RANGE_4:
+		p.aRange = 4
+	case ACCEL_RANGE_8:
+		p.aRange = 8
+	case ACCEL_RANGE_16:
+		p.aRange = 16
+	default:
+		return errors.New("invalid accelerometer FSR input")
+	}
+
+	var aConfig [1]byte
+	if err = p.read(_ACCEL_CONFIG, aConfig[:]); err != nil {
+		return err
+	}
+	aConfig[0] = (aConfig[0] & (^_AFS_SEL)) | (accRange << _AFS_SHIFT)
+
+	if err = p.write8(_ACCEL_CONFIG, aConfig[0]); err != nil {
+		return err
+	}
+	return nil
 }
 
-// SetFullScaleGyroRange allows the user to configure the scale range for the gyroscope.
-func (d Device) SetFullScaleGyroRange(rng uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), GYRO_CONFIG, []uint8{rng})
+// Sleep sets the sleep bit on the power managment 1 field.
+// When the recieved bool is true, it sets the bit to 1 thus putting
+// the peripheral in sleep mode.
+// When false is recieved the bit is set to 0 and the peripheral wakes up.
+func (p *Device) Sleep(sleepEnabled bool) (err error) {
+	// setSleepBit
+	var pwrMgt [1]byte
+	if err = p.read(_PWR_MGMT_1, pwrMgt[:]); err != nil {
+		return err
+	}
+	if sleepEnabled {
+		pwrMgt[0] = (pwrMgt[0] & (^_SLEEP_MASK)) | (1 << _SLEEP_SHIFT) // Overwrite only Sleep
+	} else {
+		pwrMgt[0] = (pwrMgt[0] & (^_SLEEP_MASK))
+	}
+	if err = p.write8(_PWR_MGMT_1, pwrMgt[0]); err != nil {
+		return err
+	}
+	return nil
 }
 
-// SetFullScaleAccelRange allows the user to configure the scale range for the accelerometer.
-func (d Device) SetFullScaleAccelRange(rng uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), ACCEL_CONFIG, []uint8{rng})
+func DefaultConfig() Config {
+	return Config{
+		AccRange:    ACCEL_RANGE_16,
+		GyroRange:   GYRO_RANGE_2000,
+		sampleRatio: 0, // TODO add const values.
+		clkSel:      0,
+	}
 }
```

Comment:
Instead of DefaultConfig, have you considered making the zero value the default config?
That way, users can just do `Configure(mpu6050.Config{GyroRange: ...})` and know that the acceleration range is also set to some reasonable default.

Reply:
I have pushed changes that make the zero value useful in latest commit!